### PR TITLE
fixes issue with typing nullish and nullable optional types

### LINKED ISSUE #110: Typesafe props not working for field that is nullable.optional or .nullish
Prop types seems to break when optional/nullable/nullish are not in the correct order. See the reprodcution below

TL;DR:

- `optional().nullable()` works
- `nullable().optional()` does not
- `nullish()` does not, as it is equivalent to `nullable().optional()`

```tsx
import { TextInput } from "@mantine/core";
import { createTsForm, useTsController } from "@ts-react/form";
import { z } from "zod";

const TextField = ({ description }: { description?: string }) => {
  const { field, error } = useTsController<string>();

  return (
    <TextInput
      value={field.value}
      onChange={(e) => field.onChange(e.target.value)}
      label={field.name}
      placeholder={description}
      error={error?.errorMessage}
    />
  );
};

const mapping = [[z.string(), TextField]] as const;

export const Form = createTsForm(mapping);

export const MyFormComponent = () => {
  return (
    <Form
      schema={z.object({
        displayNameOptionalNullable: z.string().optional().nullable(),
        displayNameNullableOptional: z.string().nullable().optional(),
        displayNameNullish: z.string().nullish(),
      })}
      props={{
        displayNameOptionalNullable: {
          // Works, no TS errors
          description: "",
        },
        displayNameNullableOptional: {
          // Does not work, Type '{ description: string; }' is not assignable to type 'undefined'
          description: "",
        },
        displayNameNullish: {
          // Does not work, Type '{ description: string; }' is not assignable to type 'undefined'
          description: "",
        },
      }}
      onSubmit={() => {
        /* ... */
      }}
    />
  );
};
```

### PR DESCRIPTION
**Pull request by iway1**
closes #110

### CODE REVIEW

**Comment by iway1** [comment]
@scamden if you have time could you check this out real quick? 

Had to mess with the unwrap types type to fix the typescript issues... also started adding some type tests

**Comment by scamden** [comment]
I'm OOO but will take a look when I'm back in a couple weeks!

File: src/__tests__/unwrap.test.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    function TextField(_: { one: string }) {
+      return <div />;
+    }
+    const mapping = [[z.string(), TextField]] as const;
+
+    const Form = createTsForm(mapping);
+
+    <Form
+      onSubmit={(data) => {
+        data.nullableString;
```
Comment (bastibuck) [inline src/__tests__/unwrap.test.tsx:16]: Do you also want to add

`data.optionalString` and
`data.nullishString` here?

**Comment by bastibuck** [comment]
If I might jump in here as well. The changes look good to me.


> also started adding some type tests

On the issue of type tests maybe this is worth taking a look at https://www.totaltypescript.com/how-to-test-your-types as this might reflect "type tests" more closely.

**Comment by scamden** [comment]
this LGTM! i literally ran into this bug with a coworker today so good time to remember to do this review ha.

**Comment by scamden** [comment]
@iway1 shall i go ahead and merge?

**Comment by iway1** [comment]
@scamden go for it! 😄

**Comment by scamden** [comment]
@iway1 great! what's the workflow for creating a release btw?

**Comment by iway1** [comment]
@scamden create a new release with the tag being the version name and it'll run and deploy the package to npm with the tag as the version (IE a tag of `1.20.5` would deploy the package with version 1.20.5)

I've just been doing auto generate for the release notes

**Comment by iway1** [comment]
For patches definitely feel free to create releases whenever you want

**Comment by scamden** [comment]
Got it :)

**Comment by iway1** [comment]
image

Success! @scamden

**Pull request merged**